src/testCommon/java/uk/gov/hmcts/darts/test/common/data/ProsecutorTestData.java

```diff
@@ -37,9 +41,14 @@ private static ProsecutorEntity createProsecutorWithCaseBasedName(int index, Cou
     }
 
     public static ProsecutorEntity createProsecutorForCaseWithName(CourtCaseEntity courtCase, String name) {
-        var prosecutor = new ProsecutorEntity();
-        prosecutor.setCourtCase(courtCase);
+        var prosecutor = someMinimalProsecutorForCase(courtCase);
         prosecutor.setName(name);
         return prosecutor;
     }
+
+    public static ProsecutorEntity someMinimalProsecutorForCase(CourtCaseEntity courtCase) {
+        var prosecutor = someMinimalProsecutor();
```

Comment:
someMinimalProsecutor also creates its own courtcase, so we are creating multiple courtcases here, woudl something like this be better? :-

```java
public static ProsecutorEntity someMinimalProsecutor() {
    return someMinimalProsecutor(createSomeMinimalCase());
}

public static ProsecutorEntity someMinimalProsecutor(CourtCaseEntity courtCase) {
    var prosecutor = new ProsecutorEntity();
    prosecutor.setCourtCase(courtCase);
    prosecutor.setName("some-prosecutor");
    var accountEntity = minimalUserAccount();
    prosecutor.setCreatedBy(accountEntity);
    prosecutor.setLastModifiedBy(accountEntity);
    return prosecutor;
}
```

Reply:
The the courtcase in from the minimal method is overwritten before being persisted when call this method.  This is the point of this pattern over the stubs which couples data generation with persistence.  minimal* is a base minimal persistence for which other methods can build upon for convenience and readability of tests.

Reply:
is what i have written not functionally teh same, but without creating superfluous entities?

Reply:
The entity is overwritten immediately so isn't causing any issues.  Having 1 place that each entity is created is desirable from a maintainability perspective.  This is another issue with the integration tests, mainly the stubs having lots of places where we new objects up.  The minimal* pattern gives an entity that is gauranteed to be persistable, so all methods should build upon that.

Reply:
@tonymort @cmagowan-hmcts I'm not sure there's a huge amount to be gained by overloading these methods here. If a case object is created, linked and overwritten then it's all done in memory and during a test run.

We would like to get these tests re-enabled as soon as possible, we should continue with what Tony has and any concerns can be raised in tickets.